x-pack/plugin/esql/qa/testFixtures/src/main/resources/match-function.csv-spec

```diff
@@ -523,3 +523,74 @@ from employees,employees_incompatible
 
 emp_no_bool:boolean
 ;
+
+testMatchWithSemanticText
+required_capability: match_function
+required_capability: semantic_text_type
+
+from semantic_text
+| where match(semantic_text_field, "something")
+| keep semantic_text_field
+| sort semantic_text_field asc
+;
+
+semantic_text_field:semantic_text
+all we have to decide is what to do with the time that is given to us
+be excellent to each other
+live long and prosper
+;
+
+testMatchWithSemanticTextAndKeyword
+required_capability: match_function
+required_capability: semantic_text_type
+
+from semantic_text
+| where match(semantic_text_field, "something") AND match(host, "host1")
+| keep semantic_text_field, host
+;
+
+semantic_text_field:semantic_text | host:keyword
+live long and prosper             | host1
+;
+
+testMatchWithSemanticTextMultiValueField
+required_capability: match_function
+required_capability: semantic_text_type
+
+from semantic_text metadata _id
+| where match(st_multi_value, "something") AND match(host, "host1")
+| keep _id, st_multi_value
+;
+
+_id: keyword | st_multi_value:semantic_text
+1            | ["Hello there!", "This is a random value", "for testing purposes"]
+;
+
+testMatchWithSemanticTextWithEvalsAndOtherFunctionsAndStats
+required_capability: match_function
+required_capability: semantic_text_type
+
+from semantic_text
+| where qstr("description:some*")
+| eval size = mv_count(st_multi_value)
+| where match(semantic_text_field, "something") AND size > 1 AND match(host, "host1")
+| STATS result = count(*)
+;
+
+result:long
+1
```

Comment:
Why not be explicit about the result returned, even with just the ID ir ? It would help understand and check that everything matches as expected?

Reply:
The test is called `testMatchWithSemanticTextWithEvalsAndOtherFunctionsAndStats` - so the use of `STATS` here is on purpose - it's not a shortcut for not checking the document ID, there are other tests that will match on actual documents. This test was added to address part of feedback I got from the initial approach for semantic search https://github.com/elastic/elasticsearch/pull/116253#discussion_r1855070008

Reply:
Thanks for explaining! 👍